sdk/core/azure-core/azure/core/utils/_pipeline_transport_rest_shared.py

```diff
@@ -287,16 +336,38 @@ def _get_raw_parts_helper(response, http_response_type: Type):
     :return: The parts of the response
     """
     body_as_bytes = response.body()
+    content_type = response.content_type
+    if body_as_bytes is None or content_type is None:
+        raise ValueError("You can't get parts if the response is not multipart/mixed")
+
     # In order to use email.message parser, I need full HTTP bytes. Faking something to make the parser happy
-    http_body = b"Content-Type: " + response.content_type.encode("ascii") + b"\r\n\r\n" + body_as_bytes
+    http_body = b"Content-Type: " + content_type.encode("ascii") + b"\r\n\r\n" + body_as_bytes
     message: Message = message_parser(http_body)
+
+    if not response.request.multipart_mixed_info:
+        raise ValueError("You can't get parts if the request is not multipart/mixed")
     requests = response.request.multipart_mixed_info[0]
+
     return response._decode_parts(message, http_response_type, requests)  # pylint: disable=protected-access
 
 
+@overload
+def _parts_helper(
+    response: _HttpResponseBackcompatMixinBase,
+) -> List[_HttpResponseBackcompatMixinBase]:
+    ...
+
+
+@overload
+def _parts_helper(
+    response: PipelineTransportHttpResponse,
+) -> List[PipelineTransportHttpResponse]:
+    ...
+
+
 def _parts_helper(
-    response: "PipelineTransportHttpResponse",
-) -> Iterator["PipelineTransportHttpResponse"]:
+    response: Any,
+) -> List[Any]:
```

Comment:
Will some of the typing for `_get_raw_parts` ( and chain of calls ) have to be updated from `Iterator` to `List` as well ?

Reply:
nvm maybe just `_get_raw_parts` as helper has been updated above